iothub/service/src/DirectMethod/DirectMethodsClient.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> to directly invoke direct methods on devices and modules in IoT hub.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods"/>
+    public class DirectMethodsClient
+    {
+        private const string DeviceMethodUriFormat = "/twins/{0}/methods";
+        private const string ModuleMethodUriFormat = "/twins/{0}/modules/{1}/methods";
+
+
+        private string _hostName;
+        private IotHubConnectionProperties _credentialProvider;
+        private HttpClient _httpClient;
+        private HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected DirectMethodsClient()
+        {
+        }
+
+        internal DirectMethodsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _credentialProvider = credentialProvider;
+            _hostName = hostName;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Interactively invokes a method on a device.
+        /// </summary>
+        /// <param name="deviceId">The device identifier for the target device.</param>
+        /// <param name="cloudToDeviceMethod">Parameters to execute a direct method on the device.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+        /// <returns>The <see cref="CloudToDeviceMethodResult"/>.</returns>
+        /// <exception cref="ArgumentNullException">When <paramref name="deviceId"/> is null.</exception>
+        /// <exception cref="ArgumentNullException">When <paramref name="cloudToDeviceMethod"/> is null.</exception>
+        /// <exception cref="ArgumentException">When <paramref name="deviceId"/> is empty or whitespace.</exception>
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods"/>
```

Comment:
```suggestion
        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-direct-methods"/>
```

MSFT docs automatically adds the appropriate region/language to the reader's view if you leave this out